### PR TITLE
Feature/upgrade to OroCommerce 6

### PR DESCRIPTION
**Pull request by DiegoJaureSynolia**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no

Updated every aspect of the plugin to Oro6 in which I include:
- Annotations -> Attributes
- ACL
- Datagrids in the oro6 way
- twig simplycity
- composer.json -> new requirements
- README.md

### CODE REVIEW

File: src/Synolia/Bundle/FavoriteBundle/Handler/FavoriteButtonAjaxHandler.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace Synolia\Bundle\FavoriteBundle\Handler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Oro\Bundle\CustomerBundle\Entity\CustomerUser;
+use Oro\Bundle\ProductBundle\Entity\Product;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Synolia\Bundle\FavoriteBundle\Entity\Favorite;
+
+final class FavoriteButtonAjaxHandler
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Handler/FavoriteButtonAjaxHandler.php:13]: not final

File: src/Synolia/Bundle/FavoriteBundle/Twig/FavoriteExtension.php [hunk 2]
```diff
@@ -4,43 +4,29 @@
...
 
-use Doctrine\ORM\EntityManager;
 use Oro\Bundle\ProductBundle\Entity\Product;
 use Oro\Bundle\ProductBundle\Model\ProductView;
 use Synolia\Bundle\FavoriteBundle\Layout\DataProvider\FavoriteDataProvider;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
-class FavoriteExtension extends AbstractExtension
+final class FavoriteExtension extends AbstractExtension
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Twig/FavoriteExtension.php:13]: not final

File: src/Synolia/Bundle/FavoriteBundle/Entity/Favorite.php [hunk 3]
```diff
@@ -6,87 +6,58 @@
...
- *          "security"={
- *              "type"="ACL",
- *              "group_name"="commerce"
- *          },
- *      }
- * )
- * @ORM\HasLifecycleCallbacks
- */
+#[ORM\Entity(repositoryClass: 'Synolia\Bundle\FavoriteBundle\Entity\Repository\FavoriteRepository')]
+#[ORM\Table(name: 'synolia_favorite')]
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Entity/Favorite.php:19]: you need a v1_1 that rename the existing table and test the upgrade between 5.1 and 6.0

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteController.php [hunk 4]
```diff
@@ -4,22 +4,27 @@
...
 
-use Oro\Bundle\LayoutBundle\Annotation\Layout;
+use Oro\Bundle\LayoutBundle\Attribute\Layout;
+use Oro\Bundle\SecurityBundle\Attribute\Acl;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Routing\Annotation\Route;
 use Synolia\Bundle\FavoriteBundle\Entity\Favorite;
 
-class FavoriteController extends AbstractController
+final class FavoriteController extends AbstractController
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteController.php:13]: not final

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+use Oro\Bundle\SecurityBundle\Authentication\TokenAccessorInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Synolia\Bundle\FavoriteBundle\Handler\FavoriteButtonAjaxHandler;
+
+/**
+ * @Route("/ajax", options={"expose"=true})
+ */
+final class FavoriteButtonAjaxController extends AbstractController
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php:18]: not final

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/config/assets.yml [hunk 6]
```diff
@@ -1,4 +1,6 @@
 styles:
     inputs:
-        - 'bundles/synoliafavorite/default/scss/variables/favorite-button-config.scss'
         - 'bundles/synoliafavorite/default/scss/components/favorite-button-ajax.scss'
+        - 'bundles/synoliafavorite/default/scss/variables/favorite-button-config.scss'
+
+    output: 'css/styles.css'
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/config/assets.yml:6]: remove the output part

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/imports/oro_product_list_item/oro_product_list_item.yml [hunk 7]
```diff
@@ -1,9 +1,19 @@
 layout:
     actions:
-        - '@setBlockTheme':
-              themes: '@SynoliaFavorite/layouts/default/imports/oro_product_list_item/oro_product_list_item.html.twig'
         - '@add':
-              id: __favorite
-              blockType: block
-              parentId: __product_image_holder
-    conditions: 'context["isFavoriteEnabled"] == true'
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/imports/oro_product_list_item/oro_product_list_item.yml:9]: i think you can keep the conditions

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/page/layout.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+layout:
+    actions:
+        - '@setBlockTheme':
+            themes: '@SynoliaFavorite/layouts/default/page/layout.html.twig'
+    conditions: 'context["isFavoriteEnabled"] == true'
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/page/layout.yml:5]: the conditions is not needed here as it's the bloc types content

**Review by lacrm — CHANGES_REQUESTED** [review]

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/imports/oro_product_list_item/oro_product_list_item.yml (hunk 7 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/imports/oro_product_list_item/oro_product_list_item.yml:9]: Condition added

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/page/layout.yml (hunk 8 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/page/layout.yml:5]: Condition deleted

File: src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/config/assets.yml (hunk 6 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Resources/views/layouts/default/config/assets.yml:6]: removed

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php (hunk 5 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php:18]: removed final part

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteController.php (hunk 4 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteController.php:13]: removed final part

File: src/Synolia/Bundle/FavoriteBundle/Twig/FavoriteExtension.php (hunk 2 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Twig/FavoriteExtension.php:13]: removed final part

File: src/Synolia/Bundle/FavoriteBundle/Handler/FavoriteButtonAjaxHandler.php (hunk 1 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Handler/FavoriteButtonAjaxHandler.php:13]: removed final part

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+use Oro\Bundle\CustomerBundle\Entity\CustomerUser;
+use Oro\Bundle\ProductBundle\Entity\Product;
+use Oro\Bundle\SecurityBundle\Authentication\TokenAccessorInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Synolia\Bundle\FavoriteBundle\Handler\FavoriteButtonAjaxHandler;
+
+/**
+ * @Route("/ajax", options={"expose"=true})
```
Comment (lacrm) [inline src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php:16]: annotation to attribute

**Review by lacrm — CHANGES_REQUESTED** [review]

File: src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php (hunk 9 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Controller/Frontend/FavoriteButtonAjaxController.php]: updated

File: src/Synolia/Bundle/FavoriteBundle/Entity/Favorite.php (hunk 3 above)
Comment (DiegoJaureSynolia) [reply src/Synolia/Bundle/FavoriteBundle/Entity/Favorite.php:19]: Updated

**Review by lacrm — APPROVED** [review]

**Pull request merged**